### ISSUE TITLE
Make system resolvers optional

### ISSUE THREAD

**Issue by obrienmd** [go-acme/lego#379]
Since https://github.com/xenolf/lego/pull/293 was merged, I've run into a problem with Caddy builds where I specifically want to check only external resolvers (i.e. Google as previously used), as internal resolvers have a few zones that direct traffic to internal resources but won't pick up external records.

Is it possible to make use of system resolvers (for now, /etc/resolv.conf, but perhaps other methods e.g. Windows DNS) optional?

mholt thinks if this was an env var, the option could easily be used in Caddy.
https://github.com/mholt/caddy/issues/1580

**Comment by obrienmd** [comment]
Also see https://github.com/mholt/caddy/issues/1580

**Comment by chrisgacsal** [comment]
It would be nice to have this feature as there is no way to use lego(Caddy) in environments which have split-horizon DNS setup.

**Comment by chrisgacsal** [comment]
@xenolf Could you please provide some input on this? It would be much appreciated. Thank you.

**Comment by mxplusb** [comment]
I also have this use case, @ldez any news on this?

I'm using this in the context of running a private Boulder/Pebble instance, so I'm trying to resolve against private DNS servers, so resolving against something public isn't an option.

**Comment by ldez** [comment]
You can define your own resolvers:

```
--dns.resolvers value        Set the resolvers to use for performing recursive DNS queries. Supported: host:port. The default is to use the system resolvers, or Google's DNS resolvers if the system's cannot be determined.
```

https://go-acme.github.io/lego/usage/cli/

**Comment by mxplusb** [comment]
I dug through the code base and [here is where you can see it in action](https://github.com/go-acme/lego/blob/master/cmd/setup_challenges.go#L107).

**Comment by ldez** [comment]
Fixes by #700

https://github.com/go-acme/lego/blob/ed14dda36192763c4db6141a99917d0f327dfdd5/challenge/dns01/nameserver.go#L65-L70
